packages/cli-kit/src/ui/executor.ts

```diff
@@ -0,0 +1,88 @@
+import {AutoComplete} from './enquirer/autocomplete.js'
+import {Input} from './enquirer/input.js'
+import {Select} from './enquirer/select.js'
+import {CustomInput} from './inquirer/input.js'
+import {CustomAutocomplete} from './inquirer/autocomplete.js'
+import {CustomSelect} from './inquirer/select.js'
+import {isTruthy} from '../environment/utilities.js'
+import {Question} from '../ui.js'
+import inquirer, {Answers, QuestionCollection} from 'inquirer'
+import enquirer from 'enquirer'
+import fuzzy from 'fuzzy'
+
+export function getMapper(): (question: Question) => unknown {
+  return isEnquirer() ? enquirerMapper : inquirerMapper
+}
+
+export async function run<
+  TName extends string & keyof TAnswers,
+  TAnswers extends {[key in TName]: string} = {[key in TName]: string},
+>(question: unknown): Promise<TAnswers> {
+  const questionName = (question as Question).name
+  if (isEnquirer()) {
+    // eslint-disable-next-line no-return-await
+    return await (question as enquirer.Prompt).run()
+  } else {
+    return (await inquirer.prompt(question as QuestionCollection<Answers>, {...(question as Question).choices}))[
+      questionName
+    ]
+  }
+}
+
+function inquirerMapper(question: Question): unknown {
+  switch (question.type) {
+    case 'input':
+    case 'password':
```

Comment:
The password type is not hiding the characters. You can check it with `dev shopify ngrok auth`.

Reply:
Ouch!. I didn't realize about it. Hope to have it ready soon